src/http/content_encoding.rs

```diff
@@ -123,22 +125,22 @@ pub fn get_prior_encoding<'a>(accept_encoding: &'a HeaderValue) -> &'static str
         .unwrap_or(IDENTITY)
 }
 
-/// Compress data.
-///
-/// # Parameters
-///
-/// * `data` - Data to be compressed.
-/// * `encoding` - Only support `br`, `gzip`, `deflate` and `identity`.
-pub fn compress(data: &[u8], encoding: &str) -> io::Result<Bytes> {
-    let mut buf = BytesMut::zeroed(4_096);
-    let read_bytes = match encoding {
-        BR => brotli::CompressorReader::new(data, 4096, 6, 20).read(&mut buf[..])?,
-        GZIP => GzEncoder::new(data, Compression::default()).read(&mut buf[..])?,
-        DEFLATE => DeflateEncoder::new(data, Compression::default()).read(&mut buf[..])?,
-        _ => return Err(io::Error::new(io::ErrorKind::Other, "Unsupported Encoding")),
-    };
-    buf.truncate(read_bytes);
-    Ok(buf.freeze())
+pub fn compress_stream(
+    input: impl Stream<Item = io::Result<Bytes>> + std::marker::Send + 'static,
```

Comment:
I believe `Send` is in the prelude.

```suggestion
    input: impl Stream<Item = io::Result<Bytes>> + Send + 'static,
```

Reply:
I shouldn't always rely on rust-analyzer and code actions, eh? 😅 fixed on 166e6e7